derive_builder_core/src/lib.rs

```diff
@@ -86,5 +86,8 @@ pub fn builder_for_struct(ast: syn::DeriveInput) -> proc_macro2::TokenStream {
 
     builder.push_build_fn(build_fn);
 
-    quote!(#builder)
+    // adds pub fn builder() to original struct
+    let extra_builder_fn = builder.add_builder_fn_to_struct(&ast);
+
+    quote!(#builder #extra_builder_fn)
```

Comment:
This absolutely must not be added as on-by-default; doing so would break any existing code that manually added an inherent method called `builder`.

There would need to be a new field on `macro_options::darling_opts::Builder` to opt into this behavior; either a `Flag`, or an `Option<Ident>` to allow overriding the name would suffice.

That added field would also be cross-checked against the `custom_constructor` field to produce a precise and helpful error message if the two had incompatible settings (this is described in more detail in another review comment)